message/tests/test_job.py

```diff
@@ -0,0 +1,121 @@
+# ############################################################################### #
+# Autoreduction Repository : https://github.com/ISISScientificComputing/autoreduce
+#
+# Copyright &copy; 2020 ISIS Rutherford Appleton Laboratory UKRI
+# SPDX - License - Identifier: GPL-3.0-or-later
+# ############################################################################### #
+"""
+Test the Message class
+"""
+import unittest
+import json
+import copy
+
+import attr
+
+from message.job import Message
+
+
+class TestMessage(unittest.TestCase):
+    """
+    Test cases for the Message class used with AMQ
+    """
+
+    def setUp(self):
+        self.empty_msg = Message()
+        self.empty_dict = {'description': None,
+                           'facility': None,
+                           'run_number': None,
+                           'instrument': None,
+                           'rb_number': None,
+                           'started_by': None,
+                           'file_path': None,
+                           'overwrite': None,
+                           'run_version': None,
+                           'job_id': None,
+                           'reduction_script': None,
+                           'reduction_arguments': None,
+                           'reduction_log': None,
+                           'admin_log': None,
+                           'return_message': None,
+                           'retry_in': None}
+        self.populated_msg = Message(run_number=11111, rb_number=22222, description='test message')
+        self.populated_dict = copy.deepcopy(self.empty_dict)
+        self.populated_dict['run_number'] = self.populated_msg.run_number
+        self.populated_dict['rb_number'] = self.populated_msg.rb_number
+        self.populated_dict['description'] = self.populated_msg.description
+        self.populated_keys = ['run_number', 'rb_number', 'description']
```

Comment:
### This comment links to the one below

You've 'setUp' various Messages and dicts which you've used in various combinations across the tests and in some cases made copies where modifications occur.

I'm just wondering why these aren't instead instantiated on-demand as and when needed by individual tests. This would avoid the need to make copies, and guarantees tests do not impact each other (e.g. modifying a variable in a test - see other comment below where I believe this occurs)

This would just mean refactoring the instantiations above into methods e.g. `create_populated_msg`, and calling these within the tests as needed (replacing any copies). `See \utils\clients\tests\test_sftp_client.py` for an example of this.

Happy to discuss

Reply:
Your suggestion is a better way of doing it given that not all tests require all the setup elements.  will make this change